README.md

```diff
@@ -1,6 +1,1864 @@
+mapping(bytes32 => bool) private s_nullifierHashes;
+
+//////////////////
+/// Constructor ///
+//////////////////
+
+constructor(address _owner, address _verifier, string memory _question) Ownable(_owner) {
+    s_question = _question;
+    /// Checkpoint 6 //////
+    i_verifier = IVerifier(_verifier);
+}
+
+//////////////////
+/// Functions ///
+//////////////////
+
+function vote(bytes memory _proof, bytes32 _nullifierHash, bytes32 _root, bytes32 _vote, bytes32 _depth) public {
+        /// Checkpoint 6 //////
+        if (s_nullifierHashes[_nullifierHash]) {
+            revert Voting__NullifierHashAlreadyUsed(_nullifierHash);
+        }
+        s_nullifierHashes[_nullifierHash] = true;
+
+        bytes32[] memory publicInputs = new bytes32[](4);
+        publicInputs[0] = _nullifierHash;
+        publicInputs[1] = _root;
+        publicInputs[2] = _vote;
+        publicInputs[3] = _depth;
+
+        if (!i_verifier.verify(_proof, publicInputs)) {
+            revert Voting__InvalidProof();
+        }
+
+        if (_vote == bytes32(uint256(1))) {
+            s_yesVotes++;
+        } else {
+            s_noVotes++;
+        }
+
+        emit VoteCast(_nullifierHash, msg.sender, _vote == bytes32(uint256(1)), block.timestamp, s_yesVotes, s_noVotes);
+    }
+```
+
+</details>
+</details>
+
+Once implemented, run your tests to make sure everything works:
+
+```javascript
+yarn test --grep "Checkpoint6"
+```
+
+### **✅ Tests Passed? You're So Close!**
+
+If your tests are green, congratulations — you’ve just completed the **core `Voting.sol` contract!** 🎉
+
+That means you now have a fully private voting system where:
+
+- ✅ Eligibility is proven via ZK proofs
+- ✅ Double-voting is prevented
+- ✅ Results are transparently tracked
+
+Run `yarn deploy` and then we’ll move on to integrating the **front-end** so users can interact with your voting app.
+
+### **🥅 Goals**
+
+- [ ] Understand how nullifiers help with preventing double-voting
+- [ ] Correctly pass proof and public inputs into the proof verification
+
+## Checkpoint 7: 🌱 Create Your Commitment – The First Step to Registration
+
+You’ve got your `Voting.sol` contract ready, and the verifier is hooked in.
+
+Now it’s time to let **real users** step into the process and bring the register button to life.
+
+But here’s the deal: before anyone can vote, they need to prove they’re registered.
+
+That starts right here, with **creating a commitment**.
+
+Think of this as the **secret handshake** for your voting system:
+
+- Each user makes their own **nullifier** (to prevent double voting)
+- Pairs it with a **secret** (their private ticket)
+- And from these, they compute a **commitment** (the only thing that goes on-chain)
+
+> 🚨 Double-check that your circuit and front-end use the **same Noir and bb versions.**
+> Mismatches will cause errors.
+> In our `nextjs/package.json`, you’ll see:
+> `"@aztec/bb.js": "0.82.0"` and `"@noir-lang/noir_js": "1.0.0-beta.3"`.
+> These must align.
```

Comment:
What does it mean to be aligned? What version with what version? Where can I check it?